marketplace-service/src/main/java/com/axonivy/market/controller/ProductController.java

```diff
@@ -113,6 +109,25 @@ public ResponseEntity<Message> syncProducts(@RequestHeader(value = AUTHORIZATION
     return new ResponseEntity<>(message, HttpStatus.OK);
   }
 
+  @PutMapping(SYNC_PRODUCT_VERSION)
+  @Operation(hidden = true)
+  public ResponseEntity<Message> syncProductVersions(@RequestHeader(value = AUTHORIZATION) String authorizationHeader) {
+    String token = getBearerToken(authorizationHeader);
+    gitHubService.validateUserOrganization(token, GitHubConstants.AXONIVY_MARKET_ORGANIZATION_NAME);
+    int nonSyncResult = metadataService.syncAllProductsMetadata();
+    var message = new Message();
+    HttpStatus statusCode = HttpStatus.OK;
+    if(nonSyncResult == 1) {
```

Comment:
this check is not helpfull, if the products is empty, we also have 0, but no error in code. INTERNAL_SERVER_ERROR it means we got an system issue during proceed request, it's different with nothing to sync